src/main/scala/com/imaginea/activegrid/core/models/ImageInfo.scala

```diff
@@ -0,0 +1,49 @@
+package com.imaginea.activegrid.core.models
+
+import com.typesafe.scalalogging.Logger
+import org.neo4j.graphdb.Node
+import org.slf4j.LoggerFactory
+
+/**
+ * Created by babjik on 22/9/16.
+ */
+
+case class ImageInfo(override val id: Option[Long]
+                     , imageId: Option[String] = None
+                     , state: Option[String] = None
+                     , ownerId: Option[String] = None
+                     , publicValue: Boolean
+                     , architecture: Option[String] = None
+                     , imageType: Option[String] = None
+                     , platform: Option[String] = None
+                     , imageOwnerAlias: Option[String] = None
+                     , name: Option[String] = None
+                     , description: Option[String] = None
+                     , rootDeviceType: Option[String] = None
+                     , rootDeviceName: Option[String] = None
+                     , version: Option[String] = None) extends BaseEntity
+
+
+object ImageInfo {
+
+  implicit class RichImageInfo(imageInfo: ImageInfo) extends Neo4jRep[ImageInfo] {
+    val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+    val label = "ImageInfo"
+
+    override def toNeo4jGraph(entity: ImageInfo): Option[Node] = {
+      logger.debug(s"toGraph for Image ${imageInfo}")
+      // TODO: Image fields
+      val map: Map[String, Any] = Map()
```

Comment:
Neo4j accept only primitive values as the persistence type.
It could be AnyVal but String is in under AnyRef. So , we made it as Any.

As per your comments in other repository I have removed this type specification in UserGroup. 
This ImageInfo entity is not started using in any of endpoints. So , forgot that changes.